acquire/bearer.go

```diff
@@ -0,0 +1,130 @@
+package acquire
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	"github.com/goph/emperror"
+)
+
+//TokenParser defines the function signature of a bearer token extractor from a payload
+type TokenParser func([]byte) (string, error)
+
+//ParseExpiration defines the function signature of a bearer token expiration date extractor
+type ParseExpiration func([]byte) (time.Time, error)
+
+//DefaultTokenParser extracts a bearer token as defined by a SimpleBearer in a payload
+func DefaultTokenParser(data []byte) (string, error) {
+	var bearer SimpleBearer
+
+	if errUnmarshal := json.Unmarshal(data, &bearer); errUnmarshal != nil {
+		return "", emperror.Wrap(errUnmarshal, "unable to parse bearer token")
+	}
+	return bearer.Token, nil
+}
+
+//DefaultExpirationParser extracts a bearer token expiration date as defined by a SimpleBearer in a payload
+func DefaultExpirationParser(data []byte) (time.Time, error) {
+	var bearer SimpleBearer
+
+	if errUnmarshal := json.Unmarshal(data, &bearer); errUnmarshal != nil {
+		return time.Time{}, emperror.Wrap(errUnmarshal, "unable to parse bearer token expiration")
+	}
+	return time.Now().Add(time.Duration(bearer.ExpiresInSeconds) * time.Second), nil
+}
+
+//RemoteBearerTokenAcquirerOptions provides configuration for the RemoteBearerTokenAcquirer
+type RemoteBearerTokenAcquirerOptions struct {
+	AuthURL        string            `json:"authURL"`
+	Timeout        time.Duration     `json:"timeout"`
+	Buffer         time.Duration     `json:"buffer"`
+	RequestHeaders map[string]string `json:"requestHeaders"`
+
+	GetToken      TokenParser
+	GetExpiration ParseExpiration
+}
+
+type remoteBearerTokenAcquirer struct {
+	options                RemoteBearerTokenAcquirerOptions
+	authValue              string
+	authValueExpiration    time.Time
+	httpClient             *http.Client
+	nonExpiringSpecialCase time.Time
+}
+
+//SimpleBearer defines the field name mappings used by the default bearer token and expiration parsers
+type SimpleBearer struct {
+	ExpiresInSeconds float64 `json:"expires_in"`
+	Token            string  `json:"serviceAccessToken"`
+}
+
+//NewRemoteBearerTokenAcquirer returns an acquirerr which fetches tokens from a configurable URL location
+//The acquirerr caches tokens and only re-fetches them from such URL once they have expired
```

Comment:
These comments become GoDocs, where paragraphs are only made if there are two enters.  These comments will read like this:
> NewRemoteBearerTokenAcquirer returns an acquirerr which fetches tokens from a configurable URL locationThe acquirerr caches tokens and only re-fetches them from such URL once they have expiredote: If you'd like for a token to never expire, set is expiration to time.Unix(0,0)

Comments should read like full sentences, because they will be put into the generated GoDoc, and if you want something to be separated into another paragraph/line, it should have an empty comment line between the two paragraphs.

Reply:
In a similar vein, I find it more difficult to read comments when the words start right next to the `//`, which is why I prefer if there is a space between them, but that's more of a personal preference.